### ISSUE TITLE
Tree view enhancements

### ISSUE THREAD

**Issue by mcarrano** [patternfly/patternfly-design#1043]
Creating this epic to cover multiple tree view enhancement requests.

**Comment by doruskova** [comment]
@mcarrano I've been working on the compact version of a tree view and I have two versions. 

- [With borders ](https://marvelapp.com/prototype/73043j4/screen/79699286) + [design spec](https://marvelapp.com/prototype/73043j4/screen/79700740)
- [Without borders](https://marvelapp.com/prototype/73043j4/screen/79700809)

The version with borders looks more organized IMHO and the advantage is that the eye doesn't focus that much on the line alignment compared to a second version without borders. I created [codepen](https://codepen.io/doruskova/pen/qBrjePv) for this version before I started with a mock because I wanted to know how difficult is to align all the lines.

The second version doesn't have borders. It looks cleaner and more simple. But, I guess that the alignment of lines might be difficult and it gives me an unfinished feeling from a visual point of view. Let me know what you think :))

**Comment by mcarrano** [comment]
@doruskova I kind of like the version without the borders better, but maybe that's just based on my preconceived notion of what this would look like. I wonder if the bordered version is too busy. What do you think @mceledonia ?

**Comment by doruskova** [comment]
@mcarrano [Updated mocks](https://marvelapp.com/prototype/73043j4/screen/79700590) - reflects the component in a context

**Comment by doruskova** [comment]
@mcarrano Updated mocks with a design specification. Includes only the version without borders and with a fill. 

Examples https://marvelapp.com/prototype/73043j4/screen/80030545 (without borders) and https://marvelapp.com/prototype/73043j4/screen/80030543 (with a fill) 
Design specification https://marvelapp.com/prototype/73043j4/section/1382710

**Comment by mcarrano** [comment]
@doruskova these look good. But are you proposing that we would support both versions? Is there a reason to have both?

@mcoker can you also take a look and let us know if you see any problems here. I think this would just be implemented as a new styling variant of the Tree View, but let me know if you see it differently.

**Comment by mcoker** [comment]
@mcarrano yep agree. It can be tricky to do this sort of thing, so we may learn more once we start, but LGTM!

**Comment by doruskova** [comment]
@mcarrano  I don't propose to support both versions. But, I hope I remember this correctly and on a PF Design share, we agreed to include both versions in Marvel for now and get feedback from @mcoker if it's feasible to implement the version without borders (because the lines need to be perfectly aligned with a title) and if it's not possible, then try to use the version with a fill because it still looks fine when the container is there.

**Comment by mcoker** [comment]
@doruskova I think both should be possible.

**Comment by mcarrano** [comment]
In that case, is the version without borders or fills the preferred choice?

**Comment by doruskova** [comment]
I'd go with the version without borders.

**Comment by mcarrano** [comment]
Have opened https://github.com/patternfly/patternfly/issues/4177 to begin implementation.
